lib/context.js

```diff
@@ -121,8 +121,14 @@ const proto = module.exports = {
       return;
     }
 
-    // unset all headers, and set those specified
-    this.res._headers = {};
+    // first unset all headers
+    if (this.res.getHeaderNames) {
+      this.res.getHeaderNames().forEach(name => this.res.removeHeader(name));
+    } else {
+      this.res._headers = {}; // Node < 8
```

Comment:
These apis has been backport to node 7.7.0 https://github.com/nodejs/node/blob/master/doc/changelogs/CHANGELOG_V7.md#7.7.0